### PR TITLE
plugins/registry_v2_scrape: support Basic authentication

### PR DESCRIPTION
**Pull request by steveej**
- [x] Depends on https://github.com/camallo/dkregistry-rs/pull/159

### CODE REVIEW

File: cincinnati/src/plugins/internal/graph_builder/release_scrape_dockerv2/registry/mod.rs [hunk 1]
```diff
@@ -209,19 +209,30 @@ pub async fn fetch_releases(
...
-        .username(username.map(ToString::to_string))
-        .password(password.map(ToString::to_string))
-        .build()
+    let registry_client = {
+        let authenticate = username.is_some() && password.is_some();
+
+        let client_builder = dkregistry::v2::Client::configure()
+            .registry(&registry.host_port_string())
+            .insecure_registry(registry.insecure)
+            .accept_invalid_certs(true);
```
Comment (lucab) [inline cincinnati/src/plugins/internal/graph_builder/release_scrape_dockerv2/registry/mod.rs]: Spurious debug line?

File: cincinnati/src/plugins/internal/graph_builder/release_scrape_dockerv2/registry/mod.rs [hunk 2]
```diff
@@ -209,19 +209,29 @@ pub async fn fetch_releases(
...
+        if authenticate {
+            client_builder
+                .username(username.map(ToString::to_string))
+                .password(password.map(ToString::to_string))
+                .build()
+                .map_err(|e| format_err!("{}", e))?
+                .authenticate(&[&format!("repository:{}:pull", &repo)])
+                .await
+        } else {
+            client_builder.build()
```
Comment (LalatenduMohanty) [inline cincinnati/src/plugins/internal/graph_builder/release_scrape_dockerv2/registry/mod.rs:228]: :+1:

**Review by LalatenduMohanty — APPROVED** [review]
/lgtm

**Pull request merged**